### PR TITLE
Generalizing transformer layers

### PR DESCRIPTION
**Pull request by AkshitaB**
* Added HF tests.
* Added docstrings with expected input shapes.

### CODE REVIEW

File: allennlp/modules/transformer/transformer_block.py
Comment (dirkgr) [inline allennlp/modules/transformer/transformer_block.py:14]: It's possible I have this terminology wrong, but why do you call this an "encoder"? What does it encode?

I would understand the terminology if this took word piece ids in its input, because those get encoded, but this takes vectors and returns vectors.

Is that what it's called in Devlin et al?

**Review by dirkgr — APPROVED** [review]

**Comment by dirkgr** [comment]
Let me know when this should get another review?

**Comment by AkshitaB** [comment]
> Let me know when this should get another review?

@dirkgr The tests are finally passing. The main changes from your last review are reducing the size of tensors in the test so that there are no memory issues, and a bunch of merge updates. I'll merge once I get another green signal from you.

File: tests/modules/transformer/toolkit_test.py
Comment (dirkgr) [inline tests/modules/transformer/toolkit_test.py:15]: ?

File: tests/modules/transformer/transformer_block_test.py
Comment (dirkgr) [inline tests/modules/transformer/transformer_block_test.py:173]: Is this fixed?

**Review by dirkgr — APPROVED** [review]
Looks like there is a bit of debug code left. Looks good otherwise!

**Pull request merged**
